### PR TITLE
Update minimal API filter docs to reflect latest changes

### PR DESCRIPTION
**Pull request by captainsafia**
Review link: https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis/min-api-filters?view=aspnetcore-7.0&branch=pr-en-us-27106

### CODE REVIEW

File: aspnetcore/fundamentals/minimal-apis/min-api-filters.md [hunk 1]
```diff
@@ -98,6 +98,18 @@ The following filter validates the `Todo` object and modifies the `Name` propert
...
+## Register a filter using an endpoint filter factory
+
+In some scenarios, it might be necessary to cache some of the information provided in the [`MethodInfo`](/dotnet/api/system.reflection.methodinfo) in a filter. For example, let's assume that we wanted to verify that the handler an endpoint filter is attached to has a first parameter that evaluates to a `Todo` type.
+
+[!code-csharp[](~/fundamentals/minimal-apis/min-api-filters/7samples/todo/Program.cs?name=snippet_filterfactory1)]
+
+In the preceding code:
+
+* The `EndpointFilterFactoryContext` object provides access to the [`MethodInfo`](/dotnet/api/system.reflection.methodinfo) associated with the endpoint's handler.
+* The signature of the handler is examined by inspecting [`MethodInfo.Parameters`](/dotnet/api/system.reflection.methodinfo.parameters) for the expected type signature. If the expect signature is found, the validation filter is registered onto the endpoint. This factory pattern is useful to register a filter that depends on the signature of the target endpoint handler.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/min-api-filters.md:110]:
```suggestion
* The signature of the handler is examined by inspecting [`MethodInfo.Parameters`](/dotnet/api/system.reflection.methodinfo.parameters) for the expected type signature. If the expected signature is found, the validation filter is registered onto the endpoint. This factory pattern is useful to register a filter that depends on the signature of the target endpoint handler.
```

File: aspnetcore/fundamentals/minimal-apis/min-api-filters.md [hunk 2]
```diff
@@ -98,6 +98,18 @@ The following filter validates the `Todo` object and modifies the `Name` propert
...
+* If a matching signature is not found, then a pass-through filter is registered.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/min-api-filters.md:111]:
```suggestion
* If a matching signature isn't found, then a pass-through filter is registered.
```

File: aspnetcore/fundamentals/minimal-apis/min-api-filters/7samples/todo/Program.cs [hunk 3]
```diff
@@ -95,6 +95,41 @@ await db.Todos.FindAsync(id)
 }).AddEndpointFilter<TodoIsValidUcFilter>();
 #endregion
 
+#region snippet_filterfactory1
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/min-api-filters/7samples/todo/Program.cs:98]: // <snippet_filterfactory1>

File: aspnetcore/fundamentals/minimal-apis/min-api-filters/7samples/todo/Program.cs [hunk 4]
```diff
@@ -95,6 +95,41 @@ await db.Todos.FindAsync(id)
...
+            if (!string.IsNullOrEmpty(validationError))
+            {
+                return Results.Problem(validationError);
+            }
+            return await next(invocationContextContext);
+        }
+    }
+    return invocationContext => next(invocationContext); 
+});
+#endregion
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/min-api-filters/7samples/todo/Program.cs:131]:
```suggestion
// </snippet_filterfactory1>
```

**Review by Rick-Anderson — APPROVED** [review]
@blowdart  has banned `#region snippet` syntax, he insists on `// </snippet_filterfactory1>`

File: aspnetcore/fundamentals/minimal-apis/min-api-filters/7samples/todo/Program.cs (hunk 3 above)
Comment (captainsafia) [reply aspnetcore/fundamentals/minimal-apis/min-api-filters/7samples/todo/Program.cs:98]:
```suggestion
// <snippet_filterfactory1>
```

**Pull request merged**